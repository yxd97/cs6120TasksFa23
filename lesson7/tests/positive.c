#include <stdio.h>
#include <stdlib.h>
// ARGS: 7
int main(int argc, char** argv) {
    if (argc < 2) {
        return 1;
    }
    int x = atoi(argv[1]);
    int r1 = x * 245;
    printf("%d x %d = %d\n", x, 245, r1);
    int r2 = x * 234;
    printf("%d x %d = %d\n", x, 234, r2);
    int r3 = x * 5;
    printf("%d x %d = %d\n", x, 5, r3);
    int r4 = x * 10;
    printf("%d x %d = %d\n", x, 10, r4);
    int r5 = x * 349562397;
    printf("%d x %d = %d\n", x, 349562397, r5);
    return 0;
}