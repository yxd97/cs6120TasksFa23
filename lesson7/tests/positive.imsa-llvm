; ModuleID = 'tests/positive.c'
source_filename = "tests/positive.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [14 x i8] c"%d x %d = %d\0A\00", align 1

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main(i32 noundef %0, ptr noundef %1) #0 {
  %3 = alloca i32, align 4
  %4 = alloca i32, align 4
  %5 = alloca ptr, align 8
  %6 = alloca i32, align 4
  %7 = alloca i32, align 4
  %8 = alloca i32, align 4
  %9 = alloca i32, align 4
  %10 = alloca i32, align 4
  %11 = alloca i32, align 4
  store i32 0, ptr %3, align 4
  store i32 %0, ptr %4, align 4
  store ptr %1, ptr %5, align 8
  %12 = load i32, ptr %4, align 4
  %13 = icmp slt i32 %12, 2
  br i1 %13, label %14, label %15

14:                                               ; preds = %2
  store i32 1, ptr %3, align 4
  br label %87

15:                                               ; preds = %2
  %16 = load ptr, ptr %5, align 8
  %17 = getelementptr inbounds ptr, ptr %16, i64 1
  %18 = load ptr, ptr %17, align 8
  %19 = call i32 @atoi(ptr noundef %18) #3
  store i32 %19, ptr %6, align 4
  %20 = load i32, ptr %6, align 4
  %21 = mul nsw i32 %20, 245
  %22 = shl i32 %20, 0
  %23 = add i32 0, %22
  %24 = shl i32 %20, 2
  %25 = add i32 %23, %24
  %26 = shl i32 %20, 4
  %27 = add i32 %25, %26
  %28 = shl i32 %20, 5
  %29 = add i32 %27, %28
  %30 = shl i32 %20, 6
  %31 = add i32 %29, %30
  %32 = shl i32 %20, 7
  %33 = add i32 %31, %32
  store i32 %33, ptr %7, align 4
  %34 = load i32, ptr %6, align 4
  %35 = load i32, ptr %7, align 4
  %36 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %34, i32 noundef 245, i32 noundef %35)
  %37 = load i32, ptr %6, align 4
  %38 = mul nsw i32 %37, 234
  %39 = shl i32 %37, 1
  %40 = add i32 0, %39
  %41 = shl i32 %37, 3
  %42 = add i32 %40, %41
  %43 = shl i32 %37, 5
  %44 = add i32 %42, %43
  %45 = shl i32 %37, 6
  %46 = add i32 %44, %45
  %47 = shl i32 %37, 7
  %48 = add i32 %46, %47
  store i32 %48, ptr %8, align 4
  %49 = load i32, ptr %6, align 4
  %50 = load i32, ptr %8, align 4
  %51 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %49, i32 noundef 234, i32 noundef %50)
  %52 = load i32, ptr %6, align 4
  %53 = mul nsw i32 %52, 5
  %54 = shl i32 %52, 0
  %55 = add i32 0, %54
  %56 = shl i32 %52, 2
  %57 = add i32 %55, %56
  store i32 %57, ptr %9, align 4
  %58 = load i32, ptr %6, align 4
  %59 = load i32, ptr %9, align 4
  %60 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %58, i32 noundef 5, i32 noundef %59)
  %61 = load i32, ptr %6, align 4
  %62 = mul nsw i32 %61, 10
  %63 = shl i32 %61, 1
  %64 = add i32 0, %63
  %65 = shl i32 %61, 3
  %66 = add i32 %64, %65
  store i32 %66, ptr %10, align 4
  %67 = load i32, ptr %6, align 4
  %68 = load i32, ptr %10, align 4
  %69 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %67, i32 noundef 10, i32 noundef %68)
  %70 = load i32, ptr %6, align 4
  %71 = mul nsw i32 %70, 34957
  %72 = shl i32 %70, 0
  %73 = add i32 0, %72
  %74 = shl i32 %70, 2
  %75 = add i32 %73, %74
  %76 = shl i32 %70, 3
  %77 = add i32 %75, %76
  %78 = shl i32 %70, 7
  %79 = add i32 %77, %78
  %80 = shl i32 %70, 11
  %81 = add i32 %79, %80
  %82 = shl i32 %70, 15
  %83 = add i32 %81, %82
  store i32 %83, ptr %11, align 4
  %84 = load i32, ptr %6, align 4
  %85 = load i32, ptr %11, align 4
  %86 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %84, i32 noundef 34957, i32 noundef %85)
  store i32 0, ptr %3, align 4
  br label %87

87:                                               ; preds = %15, %14
  %88 = load i32, ptr %3, align 4
  ret i32 %88
}

; Function Attrs: nounwind willreturn memory(read)
declare i32 @atoi(ptr noundef) #1

declare i32 @printf(ptr noundef, ...) #2

attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #1 = { nounwind willreturn memory(read) "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #2 = { "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #3 = { nounwind willreturn memory(read) }

!llvm.module.flags = !{!0, !1, !2, !3, !4}
!llvm.ident = !{!5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 8, !"PIC Level", i32 2}
!2 = !{i32 7, !"PIE Level", i32 2}
!3 = !{i32 7, !"uwtable", i32 2}
!4 = !{i32 7, !"frame-pointer", i32 2}
!5 = !{!"Ubuntu clang version 17.0.2 (++20230929073225+098e653a5bed-1~exp1~20230929073241.48)"}
