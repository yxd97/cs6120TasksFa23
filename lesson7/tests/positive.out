7 x 245 = 1715
7 x 234 = 1638
7 x 5 = 35
7 x 10 = 70
7 x 34957 = 244699
